script.js

```diff
@@ -1,4 +1,264 @@
+      deck.push(card);
+      counter = counter + 1;
+    }
+    suitIndex = suitIndex + 1;
+  }
+  return deck;
+};
+
+var obtainRandomIndex = function (size) {
+  return Math.floor(Math.random() * size);
+};
+
+var shuffleCards = function (cards) {
+  var index = 0;
+
+  while (index < cards.length) {
+    var randomIndex = obtainRandomIndex(cards.length);
+
+    var currentItem = cards[index];
+
+    var randomItem = cards[randomIndex];
+
+    cards[index] = randomItem;
+    cards[randomIndex] = currentItem;
+
+    index = index + 1;
+  }
+
+  return cards;
+};
+var deck = shuffleCards(createDeck());
+
+// Store player's cards and computer's cards in separate arrays
+var playerHand = [];
+var computerHand = [];
+
+// The maximum sum in Blackjack is 21. Here, we name a constant value in integer.
+var TWENTY_ONE = 21;
+// Dealer has to hit if their hand is below 17.
+var dealerHitLimit = 16;
+// If player has chosen to stand, then player can no longer hit until game is over.
+var playerHasChosenToStand = false;
+// Create one more variable to determine when the game ends and execute the results.
+var gameOver = false;
+
+// Deal a card to the given hand
+var dealCardToHand = function (hand) {
+  hand.push(deck.pop());
+};
+
+// Obtain sum of cards in hand
+var getHandSum = function (hand) {
+  var numOfAcesInHand = 0;
+  var sum = 0;
+  var counter = 0;
+  while (counter < hand.length) {
+    var currCard = hand[counter];
+    // Aces can be 1 or 11.
+    if (currCard.rank === 1) {
+      numOfAcesInHand += 1;
+      sum += 11;
+    } else {
+      sum += currCard.rank;
+    }
+    counter = counter + 1;
+  }
+  // Convert Aces from value of 11 to value of 1 when sum is greater than sum limit and hand contains Aces, until sum is <= to sum limit or there are no more Aces.
+  if (sum > TWENTY_ONE && numOfAcesInHand > 0) {
+    var aceCounter = 0;
+    while (aceCounter < numOfAcesInHand) {
+      sum -= 10;
+      // If the sum is less than TWENTY_ONE before converting all Ace values from 11 to 1, break out the loop and return the current sum.
+      if (sum <= TWENTY_ONE) {
+        break; // break will causes loop to finish
+      }
+      aceCounter = aceCounter + 1;
+    }
+  }
+  return sum;
+};
+
+// Create a variable to determine whether the hand contains a Blackjack combination
+var isBlackJackCombination = function (hand) {
+  return hand.length === 2 && getHandSum(hand) === TWENTY_ONE;
+};
+
+// Convert hand to a string where objects within the array are stringified
+var convertHandToString = function (hand) {
+  var cards = "Cards of ";
+  var handIndex = 0;
+
+  while (handIndex < hand.length) {
+    cards = cards + hand[handIndex].name + " " + hand[handIndex].suit;
+    if (handIndex != hand.length - 1) {
+      cards += ", ";
+    }
+    handIndex = handIndex + 1;
+  }
+  return cards;
+};
+//Hide Computer/Dealer's First Card
+var convertHandToString2 = function (hand) {
+  var cards = "Cards of " + "hidden card, ";
```

Comment:
decide if you want to use template literals or string + variable way of writing any strings/output and try to stick to it. I think right now you have both forms